### PR TITLE
Consolidate all PoET code

### PR DESCRIPTION
**Pull request by jjason**
* Moves all PoET code to sawtooth-core/consensus/poet
* Updated all module references
* Added initial packaging files
* Updated to work with run_lint and run_tests

### CODE REVIEW

File: bin/build_all [hunk 1]
```diff
@@ -35,6 +35,13 @@ docker_dir=$(cd $(dirname $(dirname $0))/docker && pwd)
 # Default args
 LANGUAGES=""
 EXCLUDE=""
+pkgs="
+    signing
+    cli
+    sdk/python
+    rest_api
+    validator
+"
```
Comment (aludvik) [inline bin/build_all]: This add shouldn't be here, the pkgs list has been move to build_python and you update that in a later commit.

File: consensus/poet/core/sawtooth_poet/__init__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2016 Intel Corporation
+# Copyright 2017 Intel Corporation
```
Comment (aludvik) [inline consensus/poet/core/sawtooth_poet/__init__.py]: Think this is in the wrong commit

File: bin/build_all [hunk 3]
```diff
@@ -41,6 +41,11 @@ pkgs="
     sdk/python
     rest_api
     validator
+    consensus/poet/cli
+    consensus/poet/common
+    consensus/poet/core
+    consensus/poet/families
+    consensus/poet/simulator
```
Comment (aludvik) [inline bin/build_all]: This should not be here. Only need to add to build_python.

File: consensus/poet/common/setup.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+# Copyright 2016 Intel Corporation
```
Comment (peterschwarz) [inline consensus/poet/common/setup.py]: 2017

File: consensus/poet/cli/setup.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
+# Copyright 2016 Intel Corporation
```
Comment (peterschwarz) [inline consensus/poet/cli/setup.py]: 2017

File: consensus/poet/families/setup.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2016 Intel Corporation
```
Comment (peterschwarz) [inline consensus/poet/families/setup.py]: 2017

File: consensus/poet/simulator/setup.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
+# Copyright 2016 Intel Corporation
```
Comment (peterschwarz) [inline consensus/poet/simulator/setup.py]: 2017

File: consensus/poet/common/setup.py (hunk 4 above)
Comment (jjason) [inline consensus/poet/common/setup.py]: D'Oh!

File: bin/build_all (hunk 1 above)
Comment (jjason) [inline bin/build_all]: Dang....it got added back when I rebased.  Will remove.

File: consensus/poet/core/sawtooth_poet/__init__.py (hunk 2 above)
Comment (jjason) [inline consensus/poet/core/sawtooth_poet/__init__.py]: Yeah....that should be in the unit test commit.  Will move.

File: consensus/poet/core/tests/unit_core.yaml [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+      - ../../../..:/project/sawtooth-core
+    working_dir: /project/sawtooth-core/consensus/poet/core
+    entrypoint: nose2-3 -s tests -v
+    environment:
+        PYTHONPATH: "\
+          /project/sawtooth-core/validator:\
+          /project/sawtooth-core/consensus/poet/common:\
+          /project/sawtooth-core/consensus/poet/core:\
+          /project/sawtooth-core/consensus/poet/core/tests:\
+          /project/sawtooth-core/consensus/poet/simulator"
```
Comment (jjason) [inline consensus/poet/core/tests/unit_core.yaml]: Same with this file....

File: consensus/poet/simulator/tests/test_simulator/unit_simulator.yaml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    working_dir: /project/sawtooth-core/consensus/poet/simulator
+    entrypoint: nose2-3 -s tests -v
+    environment:
+        PYTHONPATH: "\
+          /project/sawtooth-core/validator:\
+          /project/sawtooth-core/signing:\
+          /project/sawtooth-core/consensus/poet/common:\
+          /project/sawtooth-core/consensus/poet/core:\
+          /project/sawtooth-core/consensus/poet/simulator:\
+          /project/sawtooth-core/consensus/poet/simulator/tests"
```
Comment (jjason) [inline consensus/poet/simulator/tests/test_simulator/unit_simulator.yaml]: And this....

**Comment by jjason** [comment]
Sorry.  Forgot to mention that if you want to actually test the PR, you need to re-provision your vagrant VM.

**Comment by jsmitchell** [comment]
Tested locally. It works.

**Comment by jjason** [comment]
Pushed up a new commit.  I think I (hopefully) addressed all issues.

**Pull request merged**